crates/extensions/tedge_gen_mapper/src/js_runtime.rs

```diff
@@ -19,11 +20,19 @@ pub struct JsRuntime {
     execution_timeout: Duration,
 }
 
+static TIME_CREDITS: AtomicUsize = AtomicUsize::new(1000);
+
 impl JsRuntime {
     pub async fn try_new() -> Result<Self, LoadError> {
         let runtime = rquickjs::AsyncRuntime::new()?;
         runtime.set_memory_limit(16 * 1024 * 1024).await;
         runtime.set_max_stack_size(256 * 1024).await;
+        runtime
+            .set_interrupt_handler(Some(Box::new(|| {
+                let credits = TIME_CREDITS.fetch_sub(1, std::sync::atomic::Ordering::Relaxed);
+                credits == 0
+            })))
```

Comment:
This is not really a timeout as the frequency at which the interrupt handler is called is unknown.

- The number of credits has to be tuned. I just guessed some numbers for now.
- We could have something using a clock, though.